File: POO1.py
class Perro():
    def __init__ (self, nombre, raza):
        self.nombre = nombre
        self.raza = raza

    def comer(self):
        print(f'{self.nombre} está comiendo.')
    
    def ladrar(self):
        print(f'{self.nombre} está ladrando.')

    def mostrar_raza(self):
        print(f'{self.nombre} es de raza {self.raza}')

Mi_Perro = Perro("Frika", "Pincher")

Mi_Perro.comer()
Mi_Perro.ladrar()
Mi_Perro.mostrar_raza()

File: POO1.java
public class POO1 {

    public static void main(String[] args) {
        POO1 Perro= new POO1("Frika", "Pincher");
        Perro.comer();
        Perro.ladrar();
        Perro.MostrarRaza(); 
    }
    
    private String nombre;
    private String raza;

    public POO1(String perro, String raza) {
        this.nombre = perro;
        this.raza = raza;
    }

    public void comer() {
        System.out.println(nombre + " Está comiendo.");
    }

    public void ladrar() {
        System.out.println(nombre + " Está ladrando.");
    }

    public void MostrarRaza() {
        System.out.println(nombre + " Es de raza " + raza);
    }
}
